pkg/controllers/user/approuter/core_dns_client.go

```diff
@@ -0,0 +1,290 @@
+package approuter
```

Comment:
Can we try some info in the comments on what this controller does? Ideally all our controllers should have this sort of info.

Reply:
Sure,here is some essential info on what approuter controller does:

The dynamic DNS controller watches all the ingress resources in the cluster and creates the following DNS entries in <rancher_root_domain>:

<$ingress_name>.<$namespace>.<$cluster_id>.<rancher_root_domain> => [ingress IPs]

When an ingress resource has more than 10 IPs, only 10 IPs will be returned by DNS.

In an RKE cluster, when a node becomes unhealthy and the corresponding nginx ingress resource becomes unavailable, the dynamic DNS controller updates the DNS mapping to remove that node IP from the list.

Every once in a while (default 24h), the dynamic DNS controller will call renew to update the expiration time

Reply:
@alena1108 sure